Content.Shared/_RMC14/Vendors/SharedCMAutomatedVendorSystem.cs

```diff
@@ -129,6 +129,63 @@ protected virtual void OnVendBui(Entity<CMAutomatedVendorComponent> vendor, ref
             Dirty(actor, user);
         }
 
+        if (section.SharedSpecLimit is { } globalLimit)
+        {
+            if (HasComp<RMCVendorSpecialistComponent>(vendor))
+            {
```

Comment:
These two checks can be placed inside the same if statement to reduce indentation